invenio_app_ils/patrons/views.py

```diff
@@ -0,0 +1,87 @@
+from invenio_app_ils.circulation.search import get_loans_aggregated_by_states
+from elasticsearch import VERSION as ES_VERSION
+from invenio_app_ils.permissions import need_permissions
+from webargs import fields
+from webargs.flaskparser import use_args
+lt_es7 = ES_VERSION[0] < 7
+
+
+def get_user_loan_information_blueprint(app):
+    """Add patron views to the blueprint."""
+    blueprint = Blueprint("invenio_app_ils_patrons", __name__,)
+
+    @blueprint.route('/me/loans')
+    @use_args({"document_pid": fields.Str(required=True)})
+    def get_user_information(args):
+        return retrieve_user_information(args["document_pid"])
+    return blueprint
+
+
+def retrieve_user_loans_information(patron_pid, document_pid):
+    """Updates the information about the loans of a concrete document of the currently logged in patron."""
+    active_requested_loan_states = (
+        current_app.config["CIRCULATION_STATES_LOAN_REQUEST"]
+        + current_app.config["CIRCULATION_STATES_LOAN_ACTIVE"]
+    )
+    past_loan_states = (
+        current_app.config["CIRCULATION_STATES_LOAN_COMPLETED"]
+    )
+
+    user_information = {
+        "user_has_request": False,
+        "user_has_loan": False,
+        "user_last_loan": None,
+    }
+
+    loans_search = get_loans_aggregated_by_states(document_pid, active_requested_loan_states+past_loan_states, patron_pid)
+    # No need for the loan hits
+    loans_search = loans_search[:0]
+    loan_result = loans_search.execute()
+    for bucket in loan_result.aggregations.states.buckets:
+        if bucket["key"] in current_app.config["CIRCULATION_STATES_LOAN_ACTIVE"] \
+                and bucket["doc_count"] > 0:
+            user_information['user_has_loan'] = True
+        elif bucket["key"] in current_app.config["CIRCULATION_STATES_LOAN_REQUEST"] \
+                and bucket["doc_count"] > 0:
+            user_information['user_has_request'] = True
+        elif bucket["key"] in current_app.config["CIRCULATION_STATES_LOAN_COMPLETED"] \
+                and bucket["doc_count"] > 0:
+            search = search_by_patron_item_or_document(
+                patron_pid=patron_pid, document_pid=document_pid, filter_states=past_loan_states
+            )
+
+            search = search.sort({"end_date": {"order": "desc"}})
+            search_result = search.execute()
+            has_past_loans = search_result.hits.total > 0 if lt_es7 else search_result.hits.total.value > 0
+            if has_past_loans:
+                last_loan = search_result.hits[0]
+                user_information['user_last_loan'] = last_loan['end_date']
+
+    return user_information
+
+
+@need_permissions("patron-loans-information")
+def retrieve_user_information(document_pid):
+    """Retrieves user loan information about a concrete document."""
+    if current_user.is_authenticated:
+        user_information = retrieve_user_loans_information(current_user.id, document_pid)
+    else:
+        user_information = {
```

Comment:
My memory is really bad, did we say that this endpoint is protected only for logged in user?
You should probably remove this if then.